include/revng/Support/ErrorContainer.h

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <optional>
+
+#include "llvm/Support/Error.h"
+
+#include "revng/Support/Assert.h"
+
+class ErrorContainer {
+private:
+  std::vector<llvm::Error> Errors;
+
+public:
+  ErrorContainer() = default;
+  ErrorContainer(ErrorContainer &&EC) = default;
+  ErrorContainer &operator=(ErrorContainer &&) = default;
+
+  ~ErrorContainer() {
+    for (auto &Error : Errors) {
+      llvm::consumeError(std::move(Error));
+    }
+  }
+
+  bool empty() const { return Errors.empty(); }
+
+  void add(llvm::Error &&Error) {
+    if (Error)
+      Errors.push_back(std::move(Error));
+    else
+      llvm::consumeError(std::move(Error));
+  }
+
+  size_t count() const { return Errors.size(); }
+
+  llvm::Error *get(size_t Index) { return &Errors[Index]; }
+
+  template<typename T>
+  T fail(llvm::Error &&Error, T ReturnValue) {
+    add(std::move(Error));
+    return ReturnValue;
+  }
+
+  bool failIf(llvm::Error &&Error) {
+    if (Error) {
+      add(std::move(Error));
+      return false;
+    }
+    return true;
+  }
+
+  template<typename T>
+  T failIf(llvm::Error &&Error, T SuccessValue, T FailValue) {
+    if (Error) {
+      add(std::move(Error));
+      return FailValue;
+    }
+    return SuccessValue;
+  }
+};
+
+class ErrorContainerWrapper {
+private:
+  std::optional<ErrorContainer> EC;
+  ErrorContainer *ECP;
+
+public:
+  ErrorContainerWrapper(ErrorContainer *EC) {
+    if (EC == nullptr) {
+      this->EC = ErrorContainer();
+      ECP = &(*this->EC);
+    } else {
+      ECP = EC;
+    }
+  }
+
+  ErrorContainer &operator*() { return *ECP; }
+  ErrorContainer *operator->() { return ECP; }
+};
```

Comment:
It's a wrapper so that a `PipelineC` user can ignore errors with nullptr
e.g.
```cpp
bool rp_manager_apply_diff(..., rp_error_container *error_container) {
  // error_container might be nullptr
  ErrorContainerWrapper EC(error_container);
  // EC now either points to the error_container or to the class-variable EC
  revng_check(EC->empty());
```

Reply:
OK, doc it.

Reply:
Added a docstring atop the class, tell me if it's enough